src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const STATE_HISTORY = [];
+  const CLONED_STATE = { ...state };
+
+  for (const ACTION of actions) {
+    switch (ACTION.type) {
+      case 'addProperties':
+        for (const KEY in ACTION.extraData) {
+          CLONED_STATE[KEY] = ACTION.extraData[KEY];
```

Comment:
The variable `KEY` should follow camelCase naming convention. Consider renaming it to `key`.